DSCResources/MSFT_xAzurePackAdmin/MSFT_xAzurePackAdmin.psm1

```diff
@@ -1,157 +1,156 @@
-        $AzurePackAdminCredential,
-
-        [parameter(Mandatory = $true)]
-        [System.String]
-        $SQLServer,
-
-        [System.String]
-        $SQLInstance = "MSSQLSERVER"
-    )
-
-    if($SQLInstance -eq "MSSQLSERVER")
-    {
-        $ConnectionString = "Data Source=$SQLServer;Initial Catalog=Microsoft.MgmtSvc.Store;Integrated Security=True"
-    }
-    else
-    {
-        $ConnectionString = "Data Source=$SQLServer\$SQLInstance;Initial Catalog=Microsoft.MgmtSvc.Store;Integrated Security=True"
-    }
-
-    Invoke-Command -ComputerName . -Credential $AzurePackAdminCredential -Authentication Credssp {
-        $Ensure = $args[0]
-        $Principal = $args[1]
-        $ConnectionString = $args[2]
-        switch($Ensure)
-        {
-            "Present"
-            {
-                if(!(Get-MgmtSvcAdminUser -Principal $Principal -ConnectionString $ConnectionString))
-                {
-                    Add-MgmtSvcAdminUser -Principal $Principal -ConnectionString $ConnectionString
-                }
-            }
-            "Absent"
-            {
-                if(Get-MgmtSvcAdminUser -Principal $Principal -ConnectionString $ConnectionString)
-                {
-                    Remove-MgmtSvcAdminUser -Principal $Principal -ConnectionString $ConnectionString
-                }
-            }
-        }
-    } -ArgumentList @($Ensure,$Principal,$ConnectionString)
-
-    if(!(Test-TargetResource @PSBoundParameters))
-    {
-        throw "Set-TargetResouce failed"
-    }
-}
-
-
-function Test-TargetResource
-{
-    [CmdletBinding()]
-    [OutputType([System.Boolean])]
-    param
-    (
-        [ValidateSet("Present","Absent")]
-        [System.String]
-        $Ensure = "Present",
-
-        [parameter(Mandatory = $true)]
-        [System.String]
-        $Principal,
-
-        [parameter(Mandatory = $true)]
-        [System.Management.Automation.PSCredential]
-        $AzurePackAdminCredential,
-
-        [parameter(Mandatory = $true)]
-        [System.String]
-        $SQLServer,
-
-        [System.String]
-        $SQLInstance = "MSSQLSERVER"
-    )
-
-    $result = ((Get-TargetResource @PSBoundParameters).Ensure -eq $Ensure)
-
-    $result
-}
-
-
-Export-ModuleMember -Function *-TargetResource
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [ValidateSet("Present","Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $Principal,
+
+        [parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $AzurePackAdminCredential,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $SQLServer,
+
+        [System.String]
+        $SQLInstance = "MSSQLSERVER",
+
+        [System.Management.Automation.PSCredential]
+        $dbUser
+    )
+
+    if($SQLInstance -eq "MSSQLSERVER")

```

Comment:
Consider creating a function to create the connection string since this code is used in multiple places